doc/FormattingDecisionSpaces.md

```diff
@@ -0,0 +1,93 @@
+## Formatting Decision: Number of Spaces
+
+This is a document explaining our reasoning behind the formatting decision for the default number of spaces, representing an indentation.
+This document assumes that indentation is not aligned and that indentation requires a fixed number of spaces.
+The document discusses why we chose 4 spaces instead of 2.
+
+## Analysis
+
+Our analysis is very naive and we hope it is good enough, but we are open to contributions for a more sophisticated analysis.
+We have decided to analyse the OTP, WhatsApp code base and some others, since this was too big code bases we had access to.
+
+We decided to grep for a start of a function implementation `->` followed by a newline and 2 or 4 spaces `\s`.
+
+```sh
+# count 4 spaces
+$ pcregrep -Mr --include=".*\.erl" --include=".*\.hrl" "\->\n\s\s\s\s\w" . | grep "\->" | wc -l
+# count 2 spaces
+$ pcregrep -Mr --include=".*\.erl" --include=".*\.hrl" "\->\n\s\s\w" . | grep "\->" | wc -l
+```
+
+## Results
+
+OTP:
+  - 2 spaces: 16366
+  - 4 spaces: 113481
+
+WhatsApp:
+  - 2 spaces: x
+  - 4 spaces: 20x
+
+[Kazoo](https://github.com/2600hz/kazoo):
+  - 2 spaces: 0
+  - 4 spaces: 37355
+
+[MongooseIM](https://github.com/esl/MongooseIM):
+  - 2 spaces: 30
+  - 4 spaces: 13740
+
+[ejabberd](https://github.com/processone/ejabberd):
+  - 2 spaces: 349
+  - 4 spaces: 9010
+
+[Inaka repos](#inaka):
+  - 2 spaces: 2800
+  - 4 spaces: 1349
```

Comment:
Hm... in `elvis_core` alone I find 4329 matches (`master`).

Reply:
And if I count only 4-space blocks alone, it jumps to 9725 (but this is probably not the same measure).

Reply:
Hmmm did you run the same `pcregrep -Mr` regex or did you use another tool?

Reply:
Also did you remove your _build folders?

Reply:
I didn't use the `pcregrep`. Let me check that one.

Reply:
Ah, I see, you're counting indentation only "if after `->`". That's fair, I guess.

Reply:
Yes I needed something that would not include two x 2 spaces being confused with 4 spaces.  So there are more scenarios for that, but I thought `->` is the easiest and has least chance for confusion.